### ISSUE TITLE
Gevent workers not enabling AMQP heartbeats

### ISSUE THREAD

**Issue by seriyps** [celery/celery#1511]
![celery_gevent_heartbeat](https://f.cloud.github.com/assets/422014/976596/6639b8d6-0690-11e3-820a-7eaf8422843c.png)

As you can see in screenshot of RabbitMQ management interface, regular process-based workers (rows 1, 2, 5) enable heartbeats, but gevent worker (rows 4, 5) didn't.
The settings are equal:
```
BROKER_URL = "pyamqp://**:**@******:5672//"
BROKER_HEARTBEAT = 120.0
```
gevent worker just started with `--pool=gevent option`.

Not sure, that gevent is the root of the problem, but all other conditions are equal.

Celery v3.0.10

**Comment by thedrow** [comment]
This issue is also relevant for eventlet.
Unfortunately it was not resolved by #2057.

**Comment by ask** [comment]
Closing this, as we don't have the resources to complete this task.
